torch/csrc/autograd/functions/init.cpp

```diff
@@ -62,27 +63,83 @@ struct NoCtor {
 };
 
 template<typename C, typename T>
-static void addClass(PyObject* module, PyTypeObject& type, const char* name)
+static void addClass(PyObject* module, PyTypeObject& type, const char* name,
+  PyGetSetDef* function_properties=NULL, PyMethodDef* function_methods=NULL)
 {
-  createForwardFunctionPyTypeObject<T>(type, name);
+  createForwardFunctionPyTypeObject<T>(type, name, function_properties, function_methods);
   Py_INCREF(&type);
   PyModule_AddObject(module, name, (PyObject*)&type);
   registerCppFunction(typeid(C), &type);
 }
 
+template<typename T, typename M, typename P, M P::*ptr, typename V, PyObject* (*Convert)(V)>
+PyObject* getTupleAttr(PyObject* obj, void* _unused)
+{
+  THPCppFunction* self = (THPCppFunction*)obj;
+  auto& arr = std::static_pointer_cast<T>(self->cdata).get()->*ptr;
+  auto num_elems = arr.size();
+  THPObjectPtr py_tuple = PyTuple_New(num_elems);
+  if (!py_tuple) return NULL;
+  for (size_t i = 0; i < num_elems; ++i) {
+    PyTuple_SET_ITEM(py_tuple.get(), i, Convert(arr[i]));
+  }
+  return py_tuple.release();
```

Comment:
You might want to wrap these templates in `HANDLE_TH_ERRORS` and `END_HANDLE_TH_ERRORS`. It's going to handle C++ exceptions and convert them into Python errors.